## Problema


21 - Estás a trabalhar no ramo local `master`, e tens alguns _commits_ feitos
desde a última vez que sincronizaste com o remoto `origin`. Ao fazeres `push`
das tuas alterações, obténs a seguinte mensagem:

```
error: failed to push some refs to 'https://gitlab.com/ulht/projeto.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Explica qual é o problema e como o podes resolver, assumindo que não existem
_merge conflicts_.

## Soluções

### Solução 1

O erro que dá, deve-se ao nosso repositório local ter falta de um ou mais commits face ao repositório remoto. Para resolução deste problema, é necessário descarregar as novidades do repositório origin e junta-las ao ramo master.

```
$ git fetch origin
$ git merge origin/master
(os dois comandos acima poderão ser substituidos por $ git pull)
$ git push origin master
```

*Solução proposta por [André Santos](https://github.com/Snigy24)*
